Userland/Libraries/LibWeb/HTML/CrossOrigin/AbstractOperations.cpp

```diff
@@ -102,8 +102,17 @@ Optional<JS::PropertyDescriptor> cross_origin_get_own_property_helper(Variant<HT
         .property_key = property_key,
     };
 
+    // SameValue(e.[[Property]], P) can never be true at step 2.1 if P is not a string due to the different type, so we can return early.
+    if (!property_key.is_string()) {
+        return {};
+    }
+    auto const& property_key_string = property_key.as_string();
+
     // 2. For each e of CrossOriginProperties(O):
     for (auto const& entry : cross_origin_properties(object_const_variant)) {
+        // 2.1. If e.[[Property]] is not equal to P, then continue.
```

Comment:
Please use the exact language from the spec:
```
1. If SameValue(e.[[Property]], P) is true, then:
```

Reply:
Done.